lib/components/src/typography/elements/Link.tsx

```diff
@@ -1,13 +1,26 @@
 import React, { FunctionComponent, AnchorHTMLAttributes } from 'react';
+import { window } from 'global';
 
 export const Link: FunctionComponent<AnchorHTMLAttributes<HTMLAnchorElement>> = ({
   href: input,
   children,
   ...props
 }) => {
+  // If storybook is hosted at a non-root path (e.g. `/storybook/`),
+  // the base url needs to be prefixed to storybook paths.
+  let storybookBaseUrl = '/';
+  if (typeof window !== 'undefined') {
+    try {
+      storybookBaseUrl = window.parent.document.location.pathname;
+    } catch (error) {
+      // For composed storybooks this doesn't work due to a CORS error
+    }
```

Comment:
We should come up with a solution that works in all cases, not just some.

I'll open a discussion with @tmeasday and @shilman about how to properly handle this in all cases in docs for 7.0